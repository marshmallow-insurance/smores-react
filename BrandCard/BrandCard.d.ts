import { FC, ReactNode } from 'react';
import { MarginProps } from '../utils/space';
import { Color } from '../theme';
export type BrandCardProps = {
    /** generic card title */
    title: string;
    /** generic card body */
    body: string;
    /** card image */
    visual: string;
    /** alignment of the image */
    alignVisual?: 'left' | 'right';
    /** height of the image, leave this blank for 100% */
    visualHeight?: string;
    /** render image at bottom of card */
    visualBottom?: boolean;
    bgColour: Color;
    /** card tag */
    tag?: ReactNode;
    /** primary button */
    buttonAction?: ReactNode;
    /** fallback color scheme */
    fallbackStyle?: boolean;
    /** margin */
    maxWidth?: string;
} & MarginProps;
export declare const BrandCard: FC<BrandCardProps>;
