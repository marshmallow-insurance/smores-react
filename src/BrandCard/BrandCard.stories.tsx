import React from 'react'
import { BrandCard, BrandCardProps } from './BrandCard'
import placeHolderSvg from './assets/placeHolderImage.svg'
import { Tag } from '../Tag'
import { Button } from '../Button'

export default {
  title: 'BrandCard',
  component: BrandCard,
}

const Template = (props: BrandCardProps) => <BrandCard {...props} />

export const Default = Template.bind({})

Default.args = {
  title: 'All about marshmallow miles',
  body: 'Puzzled by your driving score? Wondering about rewards? Here’s how it all works!',
  alignVisual: 'left',
  bgColor: 'macaroon',
  visualHeight: '200px',
  visual: placeHolderSvg,
}

export const BrandCardWithImageWithTag = Template.bind({})

BrandCardWithImageWithTag.args = {
  title: 'All about marshmallow miles',
  body: 'Puzzled by your driving score? Wondering about rewards? Here’s how it all works!',
  maxWidth: '350px',
  bgColor: 'matcha',
  visualHeight: '180px',
  tag: <Tag label="default tag" bgColor="fairyFloss" color="liquorice" />,
  buttonAction: <Button fallback={true}>Tell me more</Button>,
  visual: placeHolderSvg,
}

export const BrandCardAlignImageRight = Template.bind({})

BrandCardAlignImageRight.args = {
  title: 'All about marshmallow miles',
  body: 'Puzzled by your driving score? Wondering about rewards? Here’s how it all works!',
  bgColor: 'matcha',
  alignVisual: 'right',
  visualHeight: '180px',
  buttonAction: <Button fallback={true}>Tell me more</Button>,
  visual: placeHolderSvg,
}

export const BrandCardWithImageBelow = Template.bind({})

BrandCardWithImageBelow.args = {
  title: 'All about marshmallow miles',
  body: 'Puzzled by your driving score? Wondering about rewards? Here’s how it all works!',
  maxWidth: '350px',
  visualHeight: '180px',
  alignVisual: 'left',
  visualBottom: true,
  bgColor: 'peanut',
  tag: <Tag label="new" bgColor="fairyFloss" color="liquorice" />,
  buttonAction: <Button fallback={true}>Tell me more</Button>,
  visual: placeHolderSvg,
}
