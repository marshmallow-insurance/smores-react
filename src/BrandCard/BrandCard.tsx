import React, { FC, ReactNode } from 'react'
import styled from 'styled-components'
import { MarginProps } from '../utils/space'
import { Color, theme } from '../theme'
import { Box } from '../Box'
import { Text } from '../Text'

export type BrandCardProps = {
  /** generic card title */
  title: string
  /** generic card body */
  body: string
  /** card image */
  visual: string
  /** alignment of the image */
  alignVisual?: 'left' | 'right'
  /** height of the image, leave this blank for 100% */
  visualHeight?: string
  /** render image at bottom of card */
  visualBottom?: boolean
  bgColour: Color
  /** card tag */
  tag?: ReactNode
  /** primary button */
  buttonAction?: ReactNode
  /** fallback color scheme */
  fallbackStyle?: boolean
  /** margin */
  maxWidth?: string
} & MarginProps

export const BrandCard: FC<BrandCardProps> = ({
  title,
  body,
  visual,
  visualHeight = '',
  alignVisual,
  visualBottom,
  tag,
  buttonAction,
  fallbackStyle = false,
  bgColour = 'matcha',
  maxWidth = '',
  ...otherProps
}) => {
  return (
    <Container
      maxWidth={maxWidth}
      visual={visual}
      fallbackStyle={fallbackStyle}
      bgColor={bgColour}
      {...otherProps}
    >
      {tag && <TagWrapper>{tag}</TagWrapper>}
      {visual && !visualBottom && (
        <VisualWrapper visualHeight={visualHeight}>
          <Visual
            visualUrl={visual}
            visualHeight={visualHeight}
            alignVisual={alignVisual}
          />
        </VisualWrapper>
      )}
      <Box flex alignItems="center" justifyContent="space-between" mt="16px">
        <Box flex alignItems="center">
          <Box>
            {title && (
              <Text typo="hero-alternate" color="liquorice">
                {title}
              </Text>
            )}
            {body && (
              <Text typo="body-regular" color="liquorice">
                {body}
              </Text>
            )}
          </Box>
        </Box>
      </Box>
      {buttonAction && <Box mt="16px">{buttonAction}</Box>}
      {visual && visualBottom && (
        <VisualWrapper visualHeight={visualHeight} mt="16px">
          <Visual
            visualUrl={visual}
            visualHeight={visualHeight}
            alignVisual={alignVisual}
          />
        </VisualWrapper>
      )}
    </Container>
  )
}

const Container = styled(Box)<{
  visual?: string
  fallbackStyle?: boolean
  bgColor: Color
}>`
  background: ${({ fallbackStyle, bgColor }) =>
    fallbackStyle ? theme.colors.cream : theme.colors[bgColor]};
  box-sizing: border-box;
  border-radius: 16px 16px 0px 0px;

  padding: 16px;
  position: relative;
  overflow: hidden;
`

const TagWrapper = styled(Box)`
  position: absolute;
  top: 16px;
  right: 16px;
`

const VisualWrapper = styled(Box)<{ visualHeight: string }>`
  width: 100%;

  ${({ visualHeight }) => visualHeight && `height: ${visualHeight};`}
`

const Visual = styled(Box)<{
  visualUrl: string
  visualHeight: string
  alignVisual?: string
}>`
  width: 100%;
  background-image: url('${(p) => p.visualUrl}');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: ${({ alignVisual }) =>
    alignVisual ? alignVisual : 'left'};

  ${({ visualHeight }) =>
    visualHeight ? `height: ${visualHeight};` : 'padding-top: 100%;'}
`
