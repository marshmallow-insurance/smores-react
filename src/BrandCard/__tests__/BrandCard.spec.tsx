import { render } from '@testing-library/react'
import React from 'react'

import { BrandCard } from '../BrandCard'

test('renders', () => {
  const { container } = render(
    <BrandCard
      title="Some example title"
      body="Some example body"
      visual="imgUrl"
      bgColor="tangerine"
    />,
  )
  expect(container.firstChild).toMatchSnapshot()
})
