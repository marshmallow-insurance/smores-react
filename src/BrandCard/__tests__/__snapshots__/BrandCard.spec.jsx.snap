// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<div
  class="sc-aXZVg sc-dcJsrY byxNQM kMCjAn"
>
  <div
    class="sc-aXZVg ipJMoi"
  >
    <div
      class="sc-aXZVg giPgVs"
    >
      <div
        class="sc-aXZVg byxNQM"
      />
    </div>
  </div>
</div>
`;
